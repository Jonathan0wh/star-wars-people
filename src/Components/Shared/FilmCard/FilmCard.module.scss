.card_container {
  // width: 300px;
}
