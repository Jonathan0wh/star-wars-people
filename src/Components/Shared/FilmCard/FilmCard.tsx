import { Card, Col, Skeleton, Space, Typography } from "antd";
import { useGetFilmQuery } from "services/swApi";

import styles from "./FilmCard.module.scss";

interface FilmCardProps {
  id: string;
}

function FilmCard({ id }: FilmCardProps) {
  const { Text } = Typography;

  const { isLoading, isFetching, isSuccess, data } = useGetFilmQuery(id);

  return isSuccess && data ? (
    <Col span={8}>
      <Space>
        <Card
          title={data.title}
          loading={isFetching}
          className={styles.card_container}
        >
          <Skeleton loading={isLoading} />

          <Text>{data.opening_crawl}</Text>
        </Card>
      </Space>
    </Col>
  ) : null;
}

export default FilmCard;
