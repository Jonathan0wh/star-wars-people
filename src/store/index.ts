export { store, persistor } from "./store";

export type { RootState, AppDispatch } from "./store";
