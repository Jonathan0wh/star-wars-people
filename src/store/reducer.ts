import { combineReducers } from "redux";
import { swApi } from "services/swApi";

export const rootReducer = combineReducers({
  [swApi.reducerPath]: swApi.reducer,
});
