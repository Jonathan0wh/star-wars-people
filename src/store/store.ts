import { configureStore } from "@reduxjs/toolkit";
import { setupListeners } from "@reduxjs/toolkit/query";
import { personDetailReducer } from "Components/Views/PersonDetail";
import { swApi } from "services/swApi";

export const store = configureStore({
  reducer: {
    [swApi.reducerPath]: swApi.reducer,

    personDetail: personDetailReducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(swApi.middleware),
});

setupListeners(store.dispatch);

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
