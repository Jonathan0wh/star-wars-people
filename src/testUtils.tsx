// test-utils.js
import React from "react";
import { applyMiddleware, createStore } from "redux";
import thunk from "redux-thunk";
import { Provider } from "react-redux";
import { render as rtlRender, RenderOptions } from "@testing-library/react";
// Import your own reducer
import { rootReducer } from "store/reducer";

type renderProps = {
  initialState?: object;
  store?: ReturnType<typeof createStore>;
} & RenderOptions;

function render(
  ui: React.ReactElement<any, string | React.JSXElementConstructor<any>>,
  {
    initialState,
    store = createStore(rootReducer, initialState, applyMiddleware(thunk)),
    ...renderOptions
  }: renderProps = {}
) {
  return rtlRender(ui, {
    wrapper: ({ children }: { children?: React.ReactNode }) => (
      <Provider store={store}>{children}</Provider>
    ),
    ...renderOptions,
  });
}

// re-export everything
export * from "@testing-library/react";
// override render method
export { render };
