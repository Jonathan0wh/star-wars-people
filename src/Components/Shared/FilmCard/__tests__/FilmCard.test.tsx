import { store } from "store";
import { render } from "testUtils";

import FilmCard from "../FilmCard";

describe("Shared Component: Film Card", () => {
  it("should render by default", () => {
    const component = render(<FilmCard id="1" />);

    const tree = component.asFragment();
    expect(tree).toMatchSnapshot();
  });

  it("should render with correct store data", () => {
    const component = render(<FilmCard id="1" />, { store });

    const tree = component.asFragment();
    expect(tree).toMatchSnapshot();
  });
});
