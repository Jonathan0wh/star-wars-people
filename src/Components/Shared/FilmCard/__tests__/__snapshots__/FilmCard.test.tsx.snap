// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shared Component: Film Card should render by default 1`] = `
<DocumentFragment>
  <div
    class="ant-col"
    style="flex: 0 0 300px;"
  >
    <div
      class="ant-card ant-card-bordered ant-card-hoverable"
    >
      <div
        class="ant-card-body"
      >
        <div
          class="ant-skeleton ant-skeleton-active skeleton"
        >
          <div
            class="ant-skeleton-content"
          >
            <h3
              class="ant-skeleton-title"
              style="width: 38%;"
            />
            <ul
              class="ant-skeleton-paragraph"
            >
              <li />
              <li />
              <li />
              <li />
              <li />
              <li />
              <li />
              <li
                style="width: 100%;"
              />
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Shared Component: Film Card should render with correct store data 1`] = `
<DocumentFragment>
  <div
    class="ant-col"
    style="flex: 0 0 300px;"
  >
    <div
      class="ant-card ant-card-bordered ant-card-hoverable"
    >
      <div
        class="ant-card-body"
      >
        <div
          class="ant-skeleton ant-skeleton-active skeleton"
        >
          <div
            class="ant-skeleton-content"
          >
            <h3
              class="ant-skeleton-title"
              style="width: 38%;"
            />
            <ul
              class="ant-skeleton-paragraph"
            >
              <li />
              <li />
              <li />
              <li />
              <li />
              <li />
              <li />
              <li
                style="width: 100%;"
              />
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
